### DISCUSSION TITLE
Conditional rendering in header-cell causes empty cells

### DISCUSSION THREAD

**Discussion by danielwillms** [quasarframework/quasar#14916] [General - Components / Directives / etc] [answered]
### What happened?

I wanted to use the `header-cell` Slot on the `QTable` to render different `QTh` Elements, depending on a condition.
Unfortunately this wasn't working, the first Element with `v-if` will be rendered, if the condition is true.
All other elements, whether with `v-else-if` or `v-else` condition, won't get rendered.
Wrapping it in a template causes it to render nothing at all.
I found the issue #9185 where there was an almost identical problem in the `body-cell` slot, maybe you can apply the same fix to it?

### What did you expect to happen?

Different `QTh` Elements to be rendered to their specific condition

### Reproduction URL

https://codepen.io/3m7ecc/pen/MWXvxVR

### How to reproduce?

1. Go to the provided reproduction link
2. See on Table#1 only Headers being rendered with `sortable: true`
3. See on Table#2 no headers being rendered at all

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar)

### Platforms/Browsers

Firefox, Chrome

**Comment by metalsadman** [comment]
https://codepen.io/metalsadman/pen/YzvrWgV

**Comment by danielwillms** [comment]
> https://codepen.io/metalsadman/pen/YzvrWgV

Yes, of course it will work when you put the condition inside the `QTh`, but in my case I have multiple components which use `QTh` as the root component and want to switch between those, depending on a flag set in the `columns`!

And either way, shouldn't it still be possible to use these conditions inside `header-cell`, if it's possible to use it in `body-cell`?

**Comment by metalsadman** [comment, marked as answer]
> > https://codepen.io/metalsadman/pen/YzvrWgV
> 
> Yes, of course it will work when you put the condition inside the `QTh`, but in my case I have multiple components which use `QTh` as the root component and want to switch between those, depending on a flag set in the `columns`!
> 
> And either way, shouldn't it still be possible to use these conditions inside `header-cell`, if it's possible to use it in `body-cell`?

do it like you're doing, but use a key ie. `<q-td :key="props.col.name"...`, you have a questionable condition too, where some columns you didn't define `sortable` or defined it as string so you'll have to account for that also.
https://codepen.io/metalsadman/pen/poKWEpv

**Reply by danielwillms** [reply]
The key seems to be the problem.
Now it's working, thank you very much.

The sortable thing was just for the example, in my case I extend the columns-definition with my own attributes and then decide based on the value.
